parlai/tasks/talkthewalk/worlds.py

```diff
@@ -0,0 +1,259 @@
+        self.neighborhood = choice(list(BOUNDARIES.keys()))
+        self.boundaries = [randint(0, BOUNDARIES[self.neighborhood][x])*2 for x in range(2)]
+        self.boundaries = self.boundaries + [x + 3 for x in self.boundaries]
+        self.agent_location = self._get_random_location()
+        self.target_location = self._get_random_location()
+
+    def init_sim(self, neighborhood=None, boundaries=None, start_location=None,
+            target_location=None):
+
+        if not neighborhood:
+            self.random_init()
+        else:
+            self.boundaries = boundaries
+            self.neighborhood = neighborhood
+            self.agent_location = start_location
+            self.target_location = target_location
+
+        self.landmarks, self.target_location = self.map.get_landmarks(
+                self.neighborhood,
+                self.boundaries,
+                self.target_location)
+
+    def get_agent_location(self):
+        return str((self.agent_location[0]-self.boundaries[0])*4 +
+                (self.agent_location[1]-self.boundaries[1]))
+
+    def get_text_map(self):
+        L = [y for x in zip(*self.landmarks) for y in x]
+        txt = '\n'.join([(str(i) + ':' + ' and '.join(x)) for i,x in enumerate(L)])
+        txt+= '\n' + str(self.target_location[0]*4 + self.target_location[1])+':Target'+'\n'
+        return txt
+
+    def get_current_view(self):
+        return "\n".join(['see:'+x for x in self.feature_loader.get(self.neighborhood, self.agent_location)])
+
+    def add_view_to_text(self, obs, action=None):
+        action = action or obs.get('text')
+        if action and is_action(action, forward=True):
+            obs['text'] = obs.get('text', '') + self.get_current_view() + '\n'
+
+    def execute_and_write(self, obs, action):
+        self.execute(action)
+        self.add_view_to_text(obs, action)
+
+    def execute(self, text):
+        """move the tourist"""
+        if not is_action(text):
+            return
+
+        self.agent_location = self.map.step_aware(
+                text,
+                self.agent_location,
+                self.boundaries)
+
+
+class SimulateWorld(ExecutableWorld):
+
+    boundaries = None
+    neighborhood = None
+    agent_location = None
+    target_location = None
+    guide = None
+    tourist = None
+
+
+    def __init__(self, opt, agents=None, shared=None):
+        super().__init__(opt, agents, shared)
+
+        if agents:
+            self.tourist = agents[0]
+            self.guide = agents[1]
+
+        if shared:
+            self.sim = shared['sim']
+        else:
+            self.sim = Simulator(opt)
+            self.sim.init_sim()
+
+        if agents:
+            self.send_map(self.guide)
+            self.send_location(self.tourist)
+
+    def send_map(self, agent):
+        agent.observe({'text': self.sim.get_text_map()})
+
+    def send_location(self, agent):
+        agent.observe({'text': self.sim.get_current_view()})
```

Comment:
should this be `self.sim.get_agent_location` or did you mean to send the view that the agent sees?